### PR TITLE
Merge dotnet/master to dotnet/dev/apple_crypto

### PR DESCRIPTION
**Pull request by bartonjs**
Merge master into the dev/apple_crypto feature branch to get the new engineering changes.

### CODE REVIEW

**Comment by bartonjs** [comment]
> MSBUILD : error MSB1009: Project file does not exist.

@ericstj, @weshaggard - Thoughts?  I don't see where it says *what* project doesn't exist.

**Comment by weshaggard** [comment]
@bartonjs looks like you are missing the necessary changes to build-tests.sh to parse out the useservergc option.

**Comment by bartonjs** [comment]
Looks like this is a problem of netci.groovy changing in the merge.

**Comment by bartonjs** [comment]
Test Innerloop Windows_NT Debug Build and Test please  (seeing if netci.groovy change has taken effect)

**Comment by bartonjs** [comment]
@dotnet-bot Test this please (netci.groovy change has taken effect)

**Comment by bartonjs** [comment]
Looks like a bunch of things are failing due to X509 extension OIDs not being known to macOS.  I had already fixed this in a local branch, but I guess I'll add it here since it's blocking the merge (and I probably can't add it before the merge since netci.groovy has changed in the branch now)

**Comment by bartonjs** [comment]
@ericstj or @weshaggard (or anyone): Mind looking at https://github.com/dotnet/corefx/pull/15265/commits/ac2d60a41da8bd98819ed59092c15a91dcbfff90 (the minimal "get things compiling again" commit)

@steveharter or @stephentoub (or anyone): Mind looking at https://github.com/dotnet/corefx/pull/15265/commits/7848f0ddcaef973c450bccaa6f3f843d94ad2312 and https://github.com/dotnet/corefx/pull/15265/commits/54f1e9f04e682a6b2a60b676250d174060bf7ec8? (get tests passing)

File: src/System.Security.Cryptography.Encoding/src/Configurations.props [hunk 1]
```diff
@@ -3,8 +3,9 @@
   <PropertyGroup>
     <BuildConfigurations>
       net463-Windows_NT;
+      netstandard-OSX;
```
Comment (weshaggard) [inline src/System.Security.Cryptography.Encoding/src/Configurations.props]: Is this really unique to OSX independent of the Unix build? If so then this looks correct.

Comment (bartonjs) [inline src/System.Security.Cryptography.Encoding/src/Configurations.props]: Yep, the Unix one (which includes FreeBSD/NetBSD) depends on OpenSSL, the OSX one doesn't.

**Comment by bartonjs** [comment]
I'm going to go ahead and merge this.  If nothing else 7848f0d and 54f1e9f will get reviewed during the feature branch's eventual merge to master.

**Pull request merged**